fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchReverseChainTest.java

```diff
@@ -0,0 +1,1034 @@
+    public void testCreateEncounter2() throws Exception {
+        WebTarget target = getWebTarget();
+
+        // Build a new Encounter and add reason-reference reference to procedure.
+        Encounter encounter = TestUtil.getMinimalResource(ResourceType.ENCOUNTER, Format.JSON);
+        encounter = encounter.toBuilder()
+                .status(EncounterStatus.FINISHED)
+                .reasonReference(Reference.builder().reference(of("Procedure/" + procedure2Id)).build())
+                .serviceProvider(Reference.builder().reference(of("Organization/" + organization2Id)).build())
+                .period(Period.builder().start(DateTime.of(now.minus(1, ChronoUnit.DAYS).toString())).end(DateTime.of(now.minus(1, ChronoUnit.DAYS).toString())).build())
+                .length(Duration.builder().system(Uri.of("http://unitsofmeasure.org")).code(Code.of("s")).value(Decimal.of("2" + tag)).build())
+                .type(CodeableConcept.builder().coding(Coding.builder().code(Code.of("2" + tag)).build()).build())
+                .build();
+
+        // Call the 'create' API.
+        Entity<Encounter> entity = Entity.entity(encounter, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Response response = target.path("Encounter").request().post(entity, Response.class);
+        assertResponse(response, Response.Status.CREATED.getStatusCode());
+
+        // Get the encounter's logical id value.
+        encounter2Id = getLocationLogicalId(response);
+
+        // Next, call the 'read' API to retrieve the new encounter and verify it.
+        response = target.path("Encounter/" + encounter2Id).request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+    }
+
+    @Test(groups = { "server-search-reverse-chain" }, dependsOnMethods = {"testCreateOrganization1"})
+    public void testCreateLocation() throws Exception {
+        WebTarget target = getWebTarget();
+
+        Location location = TestUtil.readExampleResource("json/spec/location-example.json");
+        location = location.toBuilder()
+                .managingOrganization(Reference.builder().reference(of("Organization/" + organization1Id)).build())
+                .build();
+
+        // Call the 'create' API.
+        Entity<Location> entity = Entity.entity(location, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Response response = target.path("Location").request().post(entity, Response.class);
+        assertResponse(response, Response.Status.CREATED.getStatusCode());
+
+        // Get the location's logical id value.
+        locationId = getLocationLogicalId(response);
+
+        // Next, call the 'read' API to retrieve the new Location and verify it.
+        response   = target.path("Location/" + locationId).request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+    }
+
+    @AfterClass
+    public void testDeleteResources() {
```

Comment:
```suggestion
    public void testDeleteResources() {
```

not a test per say?

Reply:
correct, just following pattern of one of the other tests to cleanup resources